Add positional args for directory, branch, and repository.

matro opened the pull request:
This implements `[<options>] [<directory> [<branch> [<repository]]]` but requires all `<options>` come before the positional args.

Fixes part of X1011/git-directory-deploy#14.

Commit:
Add positional args for directory, branch, and repository.

This implements '[<options>] [<directory> [<branch> [<repository]]]'
but requires all <options> come before the positional args.

matro commented:
This feel a little hacky to me, but it seems to work in my testing.

X1011 commented:
as with #19, i'm going to wait until we have tests for order of preference to merge this, cause i think the env variables will override these arguments as it is currently implemented.

Commit:
Move args-parsing to its own file.

Commit:
Add tests for args.

Commit:
Add tests for script's default settings.

Commit:
Add failing test for positional args.

matro commented:
Got some tests committed here, though I'm still working on the positional args themselves.

Commit:
Run tests in 'tests/' on CircleCI.

Commit:
Merge branch 'master' into feature/positional-args

Commit:
Remove superfluous tests file.

Commit:
Add test for positional args.

Commit:
Make  positional args to set env-vars instead of internal vars.

Commit:
Add docs for positional args.

matro commented:
I've merged master into this, and added a couple more commits. I think this is working now. All tests pass (including one specifically for the positional args), and there're docs for this.

X1011 commented on deploy.sh:
```diff
@@ -28,7 +28,14 @@ Variables:
+At the end of the command, you can optionally specify the directory, branch,
+and repository as well. Earlier values are required to specify later ones. For
+example, in order to specify <branch>, you must also specify <directory>. Like
+the command-line options, these"
```
looks like this is unfinished.

X1011 replied:
@matro so what was supposed to come after this?

matro replied:
Last minute coding before travel while tired didn't serve me here. b91b54e completes that sentence.

X1011 commented on readme.md:
```diff
@@ -37,6 +37,9 @@ Do this every time you want to deploy, or have your CI server do it.
 
 ### options
+
+`deploy.sh [--config <FILE>] [<options>] [<directory> [<branch> [<repository>]]]`
```
the flag in the script is `--config-file`, but actually, how about we change it to just `--config`?

matro replied:
Oy, good catch (and my bad). `--config` works for me.

Commit:
Use '--config' instead of '--config-file'.

Commit:
Merge branch 'master' into feature/positional-args

Commit:
Finish sentence explaining positional args.

Commit:
Make '-h' exit immediately after printing help info.

Without this, the script continues to execute, and we see this
error:

```
./deploy.sh: line 124: [: =: unary operator expected
Deploy directory '' does not exist. Aborting.
```
